packages/rs-dapi-client/src/dapi_client.rs

```diff
@@ -112,6 +112,11 @@ impl DapiClient {
             dump_dir: None,
         }
     }
+
+    /// Return the [DapiClient] address list.
+    pub fn address_list(&self) -> Arc<RwLock<AddressList>> {
+        Arc::clone(&self.address_list)
```

Comment:
It's a bad idea to clone on every call. You better return a reference and user decide either he needs to clone or not